src/dialogs/config-entry-system-options/dialog-config-entry-system-options.ts

```diff
@@ -54,27 +55,31 @@ class DialogConfigEntrySystemOptions extends LitElement {
     await this.updateComplete;
   }
 
+  public closeDialog(): void {
+    this._open = false;
+    this._error = "";
+    this._params = undefined;
+  }
+
   protected render(): TemplateResult {
     if (!this._params) {
       return html``;
     }
 
     return html`
-      <ha-paper-dialog
-        with-backdrop
-        opened
-        @opened-changed="${this._openedChanged}"
+      <ha-dialog
+        open=${this._open}
+        scrimClickAction
+        escapeKeyAction
```

Comment:
Do we want to prevent this dialog to close on `esc`, I don't think that is needed?

Reply:
When I remove these and use the `esc` key or click the modal background area to close the dialog, I am seeing a bug where I can not open it back up again without refreshing the page. Is there something else I'm missing here?

I included `scrimClickAction` and `escapeKeyAction` because I saw them used in several other areas in the codebase, but I personally prefer being able to use the `esc` key or click the modal background to close the dialog.

Reply:
Disregard, I had a typo preventing the closeDialog method from being called when the dialog was closed. This is now fixed and those two lines are removed.